iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -0,0 +1,437 @@
+                                        msg.size() + 1, // +1 for the \0 at the end
+                                        0,
+                                        (struct sockaddr*)&m_sockAddr,
+                                        sizeof(m_sockAddr));
+
+        if (sendCall.hasErrors())
+        {
+            return createErrorFromErrnum(sendCall.getErrNum());
+        }
+        else
+        {
+            return cxx::success<void>();
+        }
+    }
+}
+
+cxx::expected<std::string, IpcChannelError> UnixDomainSocket::receive()
+{
+    // we also support timedReceive. The setsockopt call sets the timeout for all further recvfrom calls, so we must set
+    // it to 0 to turn the timeout off
+    struct timeval tv;
+    tv.tv_sec = 0;
+    tv.tv_usec = 0;
+
+    return timedReceive(units::Duration(tv));
+}
+
+
+cxx::expected<std::string, IpcChannelError> UnixDomainSocket::timedReceive(const units::Duration& timeout)
+{
+    if (IpcChannelSide::CLIENT == m_channelSide)
+    {
+        std::cerr << "receiving on client side not supported for unix domain socket \"" << m_name << "\"" << std::endl;
+        return cxx::error<IpcChannelError>(IpcChannelError::INTERNAL_LOGIC_ERROR);
+    }
+
+    struct timeval tv = timeout;
+
+    auto setsockoptCall = cxx::makeSmartC(setsockopt,
+                                          cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
+                                          {static_cast<ssize_t>(ERROR_CODE)},
+                                          {EWOULDBLOCK},
+                                          m_sockfd,
+                                          SOL_SOCKET,
+                                          SO_RCVTIMEO,
+                                          (const char*)&tv,
+                                          sizeof(tv));
+
+    if (setsockoptCall.hasErrors())
+    {
+        return createErrorFromErrnum(setsockoptCall.getErrNum());
+    }
+    else
+    {
+        char message[MAX_MESSAGE_SIZE + 1]; // +1 for the \0 at the end
+
+        /// @todo extend createErrorFromErrnum with possible recvfrom errors
+        auto recvCall = cxx::makeSmartC(recvfrom,
+                                        cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
+                                        {static_cast<ssize_t>(ERROR_CODE)},
+                                        {},
+                                        m_sockfd,
+                                        &(message[0]),
+                                        MAX_MESSAGE_SIZE,
+                                        0,
+                                        nullptr,
+                                        nullptr);
+
+        if (recvCall.hasErrors())
+        {
+            // timeout is also an error and would return here
+            return createErrorFromErrnum(recvCall.getErrNum());
+        }
+        else
+        {
+            return cxx::success<std::string>(std::string(&(message[0])));
```

Comment:
- please add a 0 at message[MAX_MESSAGE_SIZE]
- you can remove the &..[0]; just "message" is sufficient